tidy3d/components/data/near2far_radiation_vectors.py

```diff
@@ -371,25 +365,30 @@ def _resample_surface_currents(
             points = np.linspace(start, stop, num_pts)
             colocation_points[idx] = points
 
-        currents = currents.colocate(*colocation_points)
-        try:
-            currents = currents.sel(f=frequency)
-        except Exception as e:
-            raise SetupError(
-                f"Frequency {frequency} not found in fields for monitor '{surface.monitor.name}'."
-            ) from e
+        for idx, points in enumerate(colocation_points):
+            if (hasattr(points, "__len__") and len(points) == 1) or not hasattr(points, "__len__"):
+                colocation_points[idx] = None
 
+        currents = currents.colocate(*colocation_points)
         return currents
 
-    # pylint:disable=too-many-locals
+    # pylint:disable=too-many-locals, too-many-arguments
     def _radiation_vectors_for_surface(
-        self, theta: ArrayLikeN2F, phi: ArrayLikeN2F, surface: Near2FarSurface, currents: xr.Dataset
+        self,
+        frequency: float,
+        theta: ArrayLikeN2F,
+        phi: ArrayLikeN2F,
+        surface: Near2FarSurface,
```

Comment:
are `Near2FarSurface` required anymore? or can this just be a `Near2FarMonitorType`?

Reply:
`Near2FarSurface` wraps around the near-field `FieldMonitor` which stores the fields to be integrated, while `Near2FarMonitorType` stores information about the far-field observation grid, so I think both are necessary for client-side n2f computation. An alternative is for `Near2FarMonitorType` to store both near field surface currents and far field radiation vectors, but that would make the monitor not atomic?

Reply:
Hm, so where I'm still a bit unclear I guess is how the linking between the near field and far field takes place. For example, looking at `test_n2f_monitors`, it seems you add some near field monitors and some far field n2f monitors and it just knows how to handle it? Could you explain this a bit more? Specifically, how each Near2FarDataType "knows" which near field data to generate its data from and where `Near2FarSurface` comes in. I thought I understood earlier but now looking at it seems I don't quite get it.

Reply:
The two possible ways to compute far fields are:

1. `RadiationVectors` --> computes radiation vectors locally from `FieldMonitor`s --> returns the appropriate `Near2FarData`.
2. Solver uses `Near2FarMonitors` to return the appropriate `Near2FarData`.

In both cases, we end up with a `Near2FarData` containing radiation vectors. Its methods can then be used to compute far fields, RCS, etc. Also, in both cases we need to supply a `Near2FarMonitor` which specifies far field angles / points for which radiation vectors are computed.